### PR TITLE
feat(platform/alz)!: add simplified private dns defaults and improve documentation

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
Adds new default values for private link DNS zones and improves documentation for the `platform/alz` library.

Removed old per-zone defaults.

### CODE REVIEW

File: platform/alz/alz_policy_default_values.json [hunk 1]
```diff
@@ -1,8 +1,45 @@
...
+          "parameter_names": [
+            "dnsZoneRegion"
+          ],
+          "policy_assignment_name": "Deploy-Private-DNS-Zones"
+        }
+      ]
+    },
     {
       "default_name": "private_dns_zone_acr",
+      "description": "DEPRECATED - please use private_dns_zone_region, private_dns_zone_resource_group_name, and private_dns_zone_subscription_id instead.",
```
Comment (jaredfholgate) [inline platform/alz/alz_policy_default_values.json:42]: Why not remove these? We are not using them and the SLZ team are not using them. Highly unlikely a customer is using them for anything production.

The policy is versioned with the library so not impact there either right?

**Review by jaredfholgate — DISMISSED** [review]
Approved with one suggestion.

File: platform/alz/alz_policy_default_values.json (hunk 1 above)
Comment (matt-FFFFFF) [reply platform/alz/alz_policy_default_values.json:42]: Removed

**Review by jaredfholgate — APPROVED** [review]
LGTM

**Pull request merged**